bitcurves/bitcurve.go

```diff
@@ -185,34 +200,35 @@ func (BitCurve *BitCurve) doubleJacobian(x, y, z *big.Int) (*big.Int, *big.Int,
 	return x3, y3, z3
 }
 
-//TODO: double check if it is okay
-// ScalarMult returns k*(Bx,By) where k is a number in big-endian form.
+// ScalarMult returns k*(Bx,By) where k is a number in big-endian form. If k=0,
+// the function returns nil, nil. This method is implemented for constant-time
+// execution, since it reads the bits of k.
 func (BitCurve *BitCurve) ScalarMult(Bx, By *big.Int, k []byte) (*big.Int, *big.Int) {
```

Comment:
This is still vulnerable to fault injection attacks. I'll address this as soon as possible.